### PR TITLE
run cypress tests as part of test script

### PR DESCRIPTION
**Pull request by christianvogt**
Fixes: https://issues.redhat.com/browse/RHOAIENG-1074

## Description

Adds new script command `cypress:ci` and runs it as part of `npm run test`

_Note: does not yet remove playwright from running._

## How Has This Been Tested?

`npm run test`

## Test Impact

Additional tests run.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
     "test:integration-ci": "npx concurrently -k -s first -n \"SB,TEST\" \"npm run build:storybook --quiet && npx http-server storybook-static --port 6006 --silent\" \"npx wait-on tcp:127.0.0.1:6006 && npm run test:accessibility && npm run test:integration\"",
     "test:e2e": "export E2E=1; playwright test ./src/__tests__/e2e",
     "storybook": "storybook dev -p 6006 -s ../config",
+    "cypress:ci": "npx concurrently -k -s first \"npm run cypress:server:build && npm run cypress:server\" \"npx wait-on tcp:127.0.0.1:9001 && npm run cypress:run:mock\"",
```
Comment (dpanshug) [inline frontend/package.json:42]: Given the pattern in our test scripts where they all start with `test:`, should we also rename this to `test:cypress-ci` for consistency?

Comment (christianvogt) [reply frontend/package.json]: updated

File: frontend/package.json [hunk 2]
```diff
@@ -27,26 +27,27 @@
...
-    "cypress:run": "cypress run -b chrome --project src/__tests__/cypress && npm run cypress:format",
+    "cypress:run": "cypress run -b chrome --project src/__tests__/cypress",
     "cypress:open:mock": "MOCK=1 npm run cypress:open",
     "cypress:run:mock": "MOCK=1 npm run cypress:run",
     "cypress:open:record": "RECORD=1 npm run cypress:open",
-    "cypress:run:record": "RECORD=1 npm run cypress:run",
+    "cypress:run:record": "RECORD=1 npm run cypress:run && npm run cypress:format",
     "cypress:server:build": "ODH_DIST_DIR=./public-cypress POLL_INTERVAL=9999999 FAST_POLL_INTERVAL=9999999 npm run build",
-    "cypress:server": "serve ./public-cypress -p 9001 -s",
+    "cypress:server": "serve ./public-cypress -p 9001 -s -L",
```
Comment (christianvogt) [inline frontend/package.json:50]: `L` option: `Do not log any request information to the console.`

File: frontend/package.json [hunk 3]
```diff
@@ -27,26 +27,27 @@
...
     "test:e2e": "export E2E=1; playwright test ./src/__tests__/e2e",
     "storybook": "storybook dev -p 6006 -s ../config",
     "cypress:open": "cypress open --project src/__tests__/cypress",
-    "cypress:run": "cypress run -b chrome --project src/__tests__/cypress && npm run cypress:format",
+    "cypress:run": "cypress run -b chrome --project src/__tests__/cypress",
     "cypress:open:mock": "MOCK=1 npm run cypress:open",
     "cypress:run:mock": "MOCK=1 npm run cypress:run",
     "cypress:open:record": "RECORD=1 npm run cypress:open",
-    "cypress:run:record": "RECORD=1 npm run cypress:run",
+    "cypress:run:record": "RECORD=1 npm run cypress:run && npm run cypress:format",
```
Comment (christianvogt) [inline frontend/package.json:48]: We only need to format after recording. 
Recording snapshots is an experimental feature that's not being used right now.

File: frontend/package.json [hunk 4]
```diff
@@ -27,26 +27,27 @@
...
-    "test": "run-s test:lint test:unit test:type-check test:integration-ci",
+    "test": "run-s test:lint test:type-check test:unit test:integration-ci test:cypress-ci",
     "test:dev": "run-s test:lint test:unit test:accessibility test:integration",
     "test:fix": "eslint --ext .js,.ts,.jsx,.tsx ./src --fix",
     "test:lint": "eslint --max-warnings 0 --ext .js,.ts,.jsx,.tsx ./src",
     "test:type-check": "tsc --noEmit",
     "test:jest": "jest",
     "test:unit": "npm run test:jest",
     "test:accessibility": "test-storybook --verbose",
+    "test:cypress-ci": "npx concurrently -k -s first \"npm run cypress:server:build && npm run cypress:server\" \"npx wait-on tcp:127.0.0.1:9001 && npm run cypress:run:mock\"",
```
Comment (christianvogt) [inline frontend/package.json:38]: Similar to `test:integration-ci`. We run the cypress build followed by the cypress server and wait for the server to be up before running cypress. The whole command ends when cypress finishes.

File: .github/workflows/pull_request.yml [hunk 5]
```diff
@@ -33,7 +33,9 @@ jobs:
         uses: actions/cache@v3
         id: frontend-cache
         with:
-          path: ${{ github.workspace }}/frontend/node_modules
+          path: |
+            ~/.cache/Cypress
+            ${{ github.workspace }}/frontend/node_modules
```
Comment (christianvogt) [inline .github/workflows/pull_request.yml:38]: We need to include cypress to cache the binary otherwise it doesn't run.

**Review by lucferbux — APPROVED** [review]
/lgtm

**Pull request merged**